### PR TITLE
Issue 73 - Basic refactor of CheckoutSummary + Added styler

### PR DESCRIPTION
**Pull request by ahmed-shehata**
Issue #73 - Basic refactor of CheckoutSummary + Added styler

### CODE REVIEW

File: AtlasUI/AtlasUI/Utils/UIView.swift [hunk 1]
```diff
@@ -0,0 +1,13 @@
+//
+//  Copyright © 2016 Zalando SE. All rights reserved.
+//
+
+import Foundation
+
+extension UIView {
+    func removeAllSubviews() {
+        for view in self.subviews {
```
Comment (bonkey) [inline AtlasUI/AtlasUI/Utils/UIView.swift]: Wouldn't be enough to just call `self.subviews.removeAll()` without any extension?

File: AtlasUI/AtlasUI/CheckoutSummaryViewController.swift [hunk 2]
```diff
@@ -46,19 +47,21 @@ final class CheckoutSummaryViewController: UIViewController, CheckoutProviderTyp
             self.setupBlurView()
             self.setupProductImageView()
             self.setupViewLabels()
-            self.setupStackView()
+            self.view.addSubview(self.stackView)
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryViewController.swift]: You have separated method for `setupProductImageView` with just addSubview, so make the same here

File: AtlasUI/AtlasUI/CheckoutSummaryViewController.swift [hunk 3]
```diff
@@ -258,49 +173,7 @@ final class CheckoutSummaryViewController: UIViewController, CheckoutProviderTyp
...
-            stackView.addSubview(topSeparatorView)
-            topSeparatorView.heightAnchor.constraintEqualToConstant(1).active = true
-            topSeparatorView.widthAnchor.constraintEqualToAnchor(stackView.widthAnchor, multiplier: 0.95).active = true
-            topSeparatorView.bottomAnchor.constraintEqualToAnchor(stackView.topAnchor).active = true
-            topSeparatorView.centerXAnchor.constraintEqualToAnchor(stackView.centerXAnchor).active = true
-        }
-    }
-
-    private func shippingView(text: String) -> UIView? {
+    internal func shippingView(text: String) -> UIView? {
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryViewController.swift]: It's used these functions only in Styler, so you should move it there, as there's no connection with controller.

But actually there should be connection, as you shouldn't put view hierarchy logic into styler, while you do have addSubview there.

So keep them here, but move addSubviews here as well.

**Comment by bonkey** [comment]
Few minor comments, otherwise good stuff.

**Comment by bonkey** [comment]
👍

File: AtlasUI/AtlasUI/CheckoutSummaryStyler.swift [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+    private func stylizeBuyButton() {
+        viewController.buyButton.translatesAutoresizingMaskIntoConstraints = false
+
+        viewController.buyButton.heightAnchor.constraintEqualToConstant(50).active = true
+        viewController.buyButton.topAnchor.constraintEqualToAnchor(viewController.termsAndConditionsButton.bottomAnchor,
+            constant: -80).active = true
+        viewController.buyButton.leadingAnchor.constraintEqualToAnchor(viewController.view.leadingAnchor, constant: 10).active = true
+        viewController.buyButton.trailingAnchor.constraintEqualToAnchor(viewController.view.trailingAnchor, constant: -10).active = true
+        let title = viewController.customer != nil ? "Buy Now" : "Connect To Zalando"
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryStyler.swift]: They should be two explicit and different buttons, not some extra conditionals here and there for actions / styling.

File: AtlasUI/AtlasUI/Utils/UIView.swift (hunk 1 above)
Comment (ahmed-shehata) [inline AtlasUI/AtlasUI/Utils/UIView.swift]: .subviews is read-only

**Comment by bonkey** [comment]
👍

**Comment by ahmed-shehata** [comment]
👍

**Pull request merged**